### DISCUSSION TITLE
React context not available in component imported with urlimports

### DISCUSSION THREAD

**Discussion by jeremydrichardson** [vercel/next.js#62944] [Help]
### Summary

I have a situation where we're trying to load a component from a url. This component uses a hook from a module in the node_modules folder that grabs the context.

For some reason the component that is imported from the urlimport can't access the context. If we use the same component but from node_modules folder, it works fine.

I've run into this issue before where there ends up being two context objects because of either multiple versions of React, or multiple versions of the hook. In this instance it doesn't appear to be that issue.

I don't understand enough of how urlimports works internally so I'm having a hard time troubleshooting. 

Does anyone know why the context would not be available in components imported from urls?

### Additional information

```js
import { Button, theme } from "my-component-library";
import { Box } from "https://url-to-box-component-module.js";

export default function Home({ layoutProps }) {
  return (
    <>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
      </Head>
      <ThemeProvider theme={theme}>
        // useTheme in <Box> component is undefined
        <Box>I am a box</Box>
        // useTheme in <Button> gets theme context object fine
        <Button variant="secondary">Go to page 2</Button>
      </ThemeProvider>
    </>
  );
}
```

**Comment by mutasim77** [comment]
1. Did you include this in your config file:
```js
module.exports = {
  experimental: {
    urlImports: ['https://your-url.com'],
  },
}
```

2. Are u using `use client` in the client components and etc?

Also if you don't mind, could you share the error?

**Reply by jeremydrichardson** [reply]
Yes, we have the urlImport key in the config.

The error is specific to the custom hook we created. But basically when we try to read the context with `useContext` it's undefined only for the `<Box>` component that is being imported via url. 

That exact same hook inside the `<Button>` component pulls the context fine and returns an object, which is the `theme` that is being passed into the `<ThemeProvider>`.

**Reply by jeremydrichardson** [reply]
I have an update with this. I realized that context is just not available during the server-side rendering of the page. If I wrap the component so it only renders on the client then it has access to the context.

Why would that be? I don't understand enough of how server rending happens with NextJS.
